packages/corpus/README.md

```diff
@@ -1,7 +1,45 @@
-[![Dart](https://github.com/dart-lang/ecosystem/actions/workflows/dart.yaml/badge.svg)](https://github.com/dart-lang/ecosystem/actions/workflows/dart.yaml)
 
-A tool to calculate the API usage for a package.
+## What's this?
 
-## Projects docs
+This is a tool to calculate the API usage for a package - what parts of a
+package's API are typically used by other Dart packages and applications.
 
-todo: add a brief project description here
+It is run from the command line:
+
+```
+dart bin/api_usage.dart <package-name>
+```
+
+It queries pub.dev for the packages that use `<package-name>`, downloads the
+referencing packages, analyzes them, and determines which portions of the target
+package's API are used. For an example usage report, see
+[collection.md](doc/collection.md).
+
+## Usage
+
+To use this tool, clone the repo, cd to `packages/corpus`, and run:
+
+```
+dart bin/api_usage.dart <package-name>
+```
+
+Some available options are:
+
+- `--package-limit`: limit the number of packages that are used for analysis
+  (this defaults to all referencing packages on pub.dev), 
+- `--show-src-references`: when there are references into a package's `lib/src/`
+  directory (something that's generally not intended to be part of a package's
+  public API), this option will include which package is using the `src/`
+  library in the output
+- `--include-old`: Include packages that haven't been published in the last
```

Comment:
Just a thought for the future. I wonder whether version numbers might be a better measure of "old" than time. More concretely, maybe we should ignore usages from packages that aren't using the latest major version of the package being queried for (or the last two major versions, or perhaps a user-supplied version number).

It might be useful to know, for example, that a new API is being used by 83% of packages that have updated to the version in which it was introduced, even though it's only used by 3% of all packages because of the number of packages that haven't yet updated to that version.

Reply:
Ah! In other words, we're currently measuring usages of A from B, even if B is pinned to using an older major version of A.

I think that's very reasonable - to just compute usages of people depending on the current major version of the package. I'll open an issue for this.